### PR TITLE
feat: plugin versions page

### PR DESCRIPTION
**Pull request by shlomomdahan**
This PR adds the plugins version by Jenkins version page.

- sidebar created similar to stats in detail page. contains plugin names, and a search box.
- when a plugin is selected, the hook will fetch the data from the raw html file found on GitHub and parse it to get the VersionData dictionary. (Keep in mind this will have to be revised once we have direct access to the data. I imagine the code will be simpler and more efficient then.
- most of the code was copied and adapted from the existing site. All functionality is retained.

Screenshot 2024-06-21 at 5 37 42 PM
Screenshot 2024-06-21 at 5 38 10 PM

Screenshot 2024-06-21 at 5 38 43 PM

Screenshot 2024-06-21 at 5 38 54 PM

### CODE REVIEW

**Comment by gounthar** [comment]
Nice, thanks!

File: src/components/PluginVersions/PluginVersionsTable.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+                                    </tr>
+                                </tfoot>
+                            </table>
+                        </div>
+                    )
+                )
+            ) : (
+                <>
+                    <h1>Plugin Versions</h1>
+                    <p>Click on a plugin to view its versions</p>
```
Comment (krisstern) [inline src/components/PluginVersions/PluginVersionsTable.tsx:74]:
```suggestion
                    <p>Please click on a plugin in the list available from the side navbar to view its versions</p>
```

Comment (krisstern) [reply src/components/PluginVersions/PluginVersionsTable.tsx:74]: Maybe we can use MUI Typography for handling text like in L73-74?: https://mui.com/material-ui/api/typography/

**Review by krisstern — CHANGES_REQUESTED** [review]

File: src/components/PluginVersions/PluginVersionsTable.tsx (hunk 1 above)
Comment (shlomomdahan) [reply src/components/PluginVersions/PluginVersionsTable.tsx]: Hi @krisstern 

I say we keep it as is for now (I changed the text) and we make the changes on the future PR when we remove sidebar and instead use a dropdown selector as discussed.

**Comment by Vandit1604** [comment]
Overall, looks very nice!
However, I see horizontal and vertical scrollbar. I guess it's because of. 

> the hook will fetch the data from the raw html file found on GitHub and parse it to get the VersionData dictionary

Although, If there's a way to get rid of the scrollbars. That would be pleasant.
Is it possible @shlomomdahan ?

**Comment by Vandit1604** [comment]
I checked it on mobile. The scrollbars are not present on smaller screens.

**Comment by shlomomdahan** [comment]
@Vandit1604  I added scrollbars to the larger tables to make it clear that there’s more data to see and to help navigate through it quicker. But if you think removing them would make the UI cleaner, I can make that change easily.

Also, I’m working on a PR to remove the sidebar as @krisstern requested. It'll replace it with a search bar and selector on the main page, along with a 'back to search' button to make it easy to start new searches.

Could we consider these updates for the next PR?

**Comment by Vandit1604** [comment]
We should take the opinion of the other mentors involved before we remove them.

> But if you think removing them would make the UI cleaner, I can make that change easily.

WDYT @krisstern @gounthar @lemeurherve about removing the scrollbar, at least for the smaller tables.

**Comment by shlomomdahan** [comment]
@Vandit1604 Oh, I didn't realize you meant you see the scrollbars on smaller tables as well. I don't see them on my end?

Can you share how it looks for you?

Thanks

Screenshot 2024-06-25 at 4 03 24 PM

**Comment by Vandit1604** [comment]
When I go to plugin version page, scrollbars are already there.

image

This is how it looks when I open a small chart

image

Could you investigate @shlomomdahan why this is happening?

**Comment by shlomomdahan** [comment]
@Vandit1604 That's strange! do you see that with the deployment link as well or just locally?

**Comment by Vandit1604** [comment]
I haven't tested this PR locally since there were some changes suggested by Kris.
I see this behavior on the Deployment link.

**Comment by shlomomdahan** [comment]
@Vandit1604 which browser are you using?

**Comment by Vandit1604** [comment]
I'm on Brave right now. On Firefox, the scrollbars are absent.
I checked, and scrollbars are also present on Chrome for me.

**Comment by shlomomdahan** [comment]
@Vandit1604 

That's very strange. I will try to see if I can reproduce the same issue. Otherwise, let's scrap the scrollbars to avoid this issue.

**Comment by krisstern** [comment]
@shlomomdahan Have you found a way to handle the issue with the scrollbars yet? This appears to be a blocker at the moment.

**Comment by shlomomdahan** [comment]
@Vandit1604 @krisstern 

I was not able to reproduce the issue on my end.  Still, I changed overflow from 'scroll' to 'auto'. Can you check if that works?

BTW - I think we should still merge this and move on. #58 will likely fix this issue.

**Comment by krisstern** [comment]
Let's make sure we review this PR properly before merging prematurely. 

@Vandit1604 Please see if you could reproduce the issue still.

**Review by Vandit1604 — APPROVED** [review]
The scrollbar issue has been resolved for me. I have checked on Brave, Firefox and Chrome. LGTM now

**Comment by krisstern** [comment]
Thanks @Vandit1604!

**Pull request merged**